File: RiturajGupta21/SelectionSort.py
def sort(arr):
    for i in range(len(arr)-1, 0, -1):
        pos = 0
        for j in range(1, i+1):
            if arr[j] > arr[pos]:
                pos = j
        temp = arr[i]
        arr[i] = arr[pos]
        arr[pos] = temp
    return arr


print(sort([1, 3, 2, 4, 6, 5]))

File: AM-ash-OR-AM-I/csw/sorting/SelectionSort.java
package sorting;

import utils.Util;

public class SelectionSort {
    static void selectionSort(int[] arr) {
        for (int i = 0; i < arr.length; i++) {
            int minPos = i;
            for (int j = i; j < arr.length; j++) {
                if (arr[j] < arr[minPos]) {
                    minPos = j;
                }
            }
            Util.swap(arr, i, minPos);
        }
    }

    public static void main(String[] args) {
        int[] arr = { 4324, 78, 5435, 354, 1, 324, 34 };
        selectionSort(arr);
        Util.display(arr, "Selection sort array: ");
    }
}
